OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -1470,7 +1470,35 @@ static class DMOpcodeHandlers {
             if (value.TryGetValueAsString(out string refString)) {
                 if(int.TryParse(refString, out var refID))
                 {
-                    state.Push(new DreamValue(DreamObject.GetFromReferenceID(state.DreamManager, refID)));
+                    // The first digit is the type
+                    var typeId = int.Parse(refString.Substring(0, 1));
+                    refID = int.Parse(refString.Substring(1));
+
+                    switch (typeId)
+                    {
+                        // DreamObject
+                        case 1:
+                        {
+                            var obj = DreamObject.GetFromReferenceID(state.DreamManager, refID);
+                            state.Push(obj is null ? DreamValue.Null : new DreamValue(obj));
```

Comment:
`new DreamValue(null)` will create a value equivalent to `DreamValue.Null`
```suggestion
                            state.Push(new DreamValue(obj));
```